parachain/primitives/router/src/export.rs

```diff
@@ -0,0 +1,1121 @@
+			log::trace!(target: "ethereum_blob_exporter", "skipped due to unmatched remote destination {dest:?}.");
+			return Err(SendError::NotApplicable)
+		}
+
+		let (local_net, local_sub) = universal_source
+			.take()
+			.ok_or_else(|| {
+				log::error!(target: "ethereum_blob_exporter", "universal source not provided.");
+				SendError::MissingArgument
+			})?
+			.split_global()
+			.map_err(|()| {
+				log::error!(target: "ethereum_blob_exporter", "could not get global consensus from universal source '{universal_source:?}'.");
+				SendError::Unroutable
+			})?;
+
+		if local_net != RelayNetwork::get() {
+			log::trace!(target: "ethereum_blob_exporter", "skipped due to unmatched relay network {local_net:?}.");
+			return Err(SendError::NotApplicable)
+		}
+
+		let para_id = match local_sub {
+			X1(Parachain(para_id)) => para_id,
+			_ => {
+				log::error!(target: "ethereum_blob_exporter", "could not get parachain id from universal source '{local_sub:?}'.");
+				return Err(SendError::MissingArgument)
+			},
+		};
+
+		let message = message.take().ok_or_else(|| {
+			log::error!(target: "ethereum_blob_exporter", "xcm message not provided.");
+			SendError::MissingArgument
+		})?;
+
+		let mut converter = XcmConverter::new(&message, &bridged_network);
+		let (payload, max_target_fee) = converter.convert().map_err(|err|{
+			log::error!(target: "ethereum_blob_exporter", "unroutable due to pattern matching error '{err:?}'.");
+			SendError::Unroutable
+		})?;
+
+		if max_target_fee.is_some() {
+			log::error!(target: "ethereum_blob_exporter", "unroutable due not supporting max target fee.");
+			return Err(SendError::Unroutable)
+		}
+
+		let (encoded, handler) = payload.abi_encode();
+
+		let blob = BridgeMessage(para_id.into(), handler, encoded).encode();
+		let hash: [u8; 32] = sp_io::hashing::blake2_256(blob.as_slice());
+
+		log::info!(target: "ethereum_blob_exporter", "message validated {hash:#?}.");
+
+		// TODO: Fees if any currently returning empty multi assets as cost
+		Ok(((blob, hash), MultiAssets::default()))
+	}
+
+	fn deliver((blob, hash): (Vec<u8>, XcmHash)) -> Result<XcmHash, SendError> {
+		let mut blob = blob.clone();
+		let mut input: &[u8] = blob.as_mut();
+		let BridgeMessage(source_id, handler, payload) = BridgeMessage::decode(&mut input)
+			.map_err(|err| {
+				log::trace!(target: "ethereum_blob_exporter", "undeliverable due to decoding error '{err:?}'.");
+				SendError::NotApplicable
+			})?;
+		Submitter::submit(&source_id, handler, payload.as_ref()).map_err(|err| {
+			log::error!(target: "ethereum_blob_exporter", "undeliverable due to submitter error '{err:?}'.");
+			SendError::Unroutable
+		})?;
+		log::info!(target: "ethereum_blob_exporter", "message delivered {hash:#?}.");
+		Ok(hash)
+	}
+}
+
+#[derive(PartialEq, RuntimeDebug)]
+enum NativeTokens {
+	Unlock { asset: H160, destination: H160, amount: u128 },
+}
+
+#[derive(PartialEq, RuntimeDebug)]
+enum OutboundPayload {
+	NativeTokens(NativeTokens),
+}
```

Comment:
Every type of message we send out will be described in this enum. 

Conversion will flow like:
```
Xcm<()> -> OutboundPayload -> (payload, handler)
```